pkg/zinit/commands.go

```diff
@@ -304,3 +304,27 @@ func (c *Client) Forget(service string) error {
 func (c *Client) Kill(service string, sig Signal) error {
 	return c.cmd(fmt.Sprintf("kill %s %s", service, string(sig)), nil)
 }
+
+// Terminate a service completely (stop, forget and remove) if it matches an exec pattern
+func (c *Client) Terminate(service string, execPattern string) error {
```

Comment:
I have some concerns about this method. It's very use case specific that it shouldn't be part of the client. I mean there is no actual use case for it outside this PR. Hence i suggest that we need to either generalize the use case, or move it outside of the client.

Of course generalizing the use case is the best thing to do. And i suggest the following "interface"

```go
func (c *Client) Destroy(filter ...Filter) error {
   // get all services, and for each service that matches ALL the filters, 
   // apply the destroy. operation
}

type Filter interface{
    matches(service *Service) bool
}

type nameFilter struct{
  name string
  }
// implement FIlter for the nameFilter type
// add other filter like `exe` filter  

// then u can call the Destroy like this
client.Destoy(WithName("dhcp-zos"), WithExe("udhcpc"))

// also if filters list is empty the method should return an error that search pattern is too wide
```